tracer/src/Datadog.Trace/IAST/Aspects/StringAspect.cs

```diff
@@ -0,0 +1,246 @@
+        /// <param name="param4">fourth parameter</param>
+        /// <returns>target concatenated to params</returns>
+        [AspectMethodReplace("System.String::Concat(System.String,System.String,System.String,System.String,System.String)", AspectFilter.StringLiterals)]
+        public static string Concat(string target, string param1, string param2, string param3, string param4)
+        {
+            return Concat_Internal(new string[] { target, param1, param2, param3, param4 });
+        }
+
+        /// <summary>
+        /// String Concat with four objects
+        /// </summary>
+        /// <param name="target">main string instance</param>
+        /// <param name="param1">first parameter</param>
+        /// <param name="param2">second parameter</param>
+        /// <param name="param3">third parameter</param>
+        /// <param name="param4">fourth parameter</param>
+        /// <returns>target concatenated to params</returns>
+        [AspectMethodReplace("System.String::Concat(System.Object,System.Object,System.Object,System.Object,System.Object)")]
+        public static string Concat(object target, object param1, object param2, object param3, object param4)
+        {
+            return Concat_Internal(new string[] { target?.ToString(), param1?.ToString(), param2?.ToString(), param3?.ToString(), param4?.ToString() });
+        }
+
+        /// <summary>
+        /// Concat of all elements in an object array
+        /// </summary>
+        /// <param name="values">the object array</param>
+        /// <returns>target concatenated to params</returns>
+        [AspectMethodReplace("System.String::Concat(System.Object[])")]
+        public static string Concat(object[] values)
+        {
+            return Concat_Internal(Array.ConvertAll(values, x => x?.ToString()));
+        }
+
+        /// <summary>
+        /// Concat of all elements in a string array
+        /// </summary>
+        /// <param name="values">the string array</param>
+        /// <returns>target concatenated to params</returns>
+        [AspectMethodReplace("System.String::Concat(System.String[])")]
+        public static string Concat(string[] values)
+        {
+            return Concat_Internal(values);
+        }
+
+        /// <summary>
+        /// Concat of all elements in an object IEnumerable
+        /// </summary>
+        /// <param name="values">the object array</param>
+        /// <returns>target concatenated to params</returns>
+        [AspectMethodReplace("System.String::Concat(System.Collections.Generic.IEnumerable`1<System.String>)")]
+        [AspectMethodReplace("System.String::Concat(System.Collections.Generic.IEnumerable`1<!!0>)")]
+        public static string Concat(object values)
+        {
+            return (Concat_Internal(GetArrayIfEnumerable(values)));
+        }
+
+        private static string Concat_Internal(string[] values, int taintedCandidateIndex = -1)
+        {
+            Console.WriteLine();
```

Comment:
Are `Console.Writeline` calls expected here?

Reply:
No. This is completely temporary to demonstrate the aspect did something. Removed.